fulp_modules/main_features/bloodsuckers/code/bloodsucker/bloodsucker_crypt.dm

```diff
@@ -602,6 +613,121 @@
 			return
 
 
+/obj/structure/bloodsucker/vassalrack/proc/shapeshift_victim(mob/living/user, mob/living/target)
+	var/datum/antagonist/bloodsucker/bloodsuckerdatum = user.mind.has_antag_datum(/datum/antagonist/bloodsucker)
+	/// To deal with Blood
+	var/mob/living/carbon/human/C = user
+	var/mob/living/carbon/human/H = target
+
+	/// Dead? No worries! Shift them into one of our creations!
+	if(H.stat == DEAD)
+		to_chat(user, "<span class='notice'>Do you wish to rebuild this body as something graeter? This will remove any restraints they might have, make them into a mute husk, and will cost 150 Blood!</span>")
+		var/list/revive_options = list(
+			"Yes" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_yes"),
+			"No" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_no")
+			)
+		var/revive_response = show_radial_menu(user, src, revive_options, radius = 36, require_near = TRUE)
+		switch(revive_response)
+			if("Yes")
+				to_chat(user, "<span class='danger'>You have brought [target] back from the Dead!</span>")
+				to_chat(target, "<span class='announce'>As Blood drips over your body, your heart begins to beat... You live again!</span>")
+				C.blood_volume -= 150
+				target.revive(full_heal = TRUE, admin_revive = TRUE)
+				ADD_TRAIT(target, TRAIT_MUTE, BLOODSUCKER_TRAIT)
+				H.become_husk()
+				return
+			else
+				to_chat(user, "<span class='danger'>You decide not to revive [target].</span>")
+				/// Unbuckle them now.
```

Comment:
again, this is an autodoc comment that also has no reason for existing, anyone reading this KNOWS that the next line will unbuckle the mob